src/tools/perf/lib/libperf.c

```diff
@@ -1188,22 +1311,29 @@ static ucs_status_t ucp_perf_test_setup_endpoints(ucx_perf_context_t *perf,
     ucs_status_t status;
     unsigned i;
 
-    /* pack the local endpoints data and send to the remote peer */
-    status = ucp_perf_test_send_local_data(perf, features);
-    if (status != UCS_OK) {
-        goto err;
-    }
+    if (perf->params.flags & UCX_PERF_TEST_FLAG_LOOPBACK) {
+        status = ucp_perf_test_loopback_ep_create(perf, features);
+        if (status != UCS_OK) {
+            goto err;
+        }
+    } else {
```

Comment:
there is still code duplication
since we provide RTE which can communicate with itself why do we need a special function "ucp_perf_test_loopback_ep_create"? i would expect that ucp_perf_test_send_local_data+ucp_perf_test_receive_remote_data will also work in loopback case, maybe with slight changes